prime-dotnet-webapi/Models/SiteRegistration/Site.cs

```diff
@@ -84,5 +84,33 @@ public IEnumerable<DayOfWeek> DaysOpen(DateTimeOffset? atTime = null)
                 .Select(b => b.Day)
                 .Distinct();
         }
+
+        public SiteStatus AddStatus(SiteStatusType siteStatusType)
+        {
+            var newStatus = SiteStatus.FromType(siteStatusType, Id);
+
+            if (SiteStatuses == null)
+            {
+                SiteStatuses = new List<SiteStatus>();
+            }
+            SiteStatuses.Add(newStatus);
+
+            return newStatus;
+        }
+
+        /// <summary>
+        /// Gets the most recent Status of the Site.
+        /// </summary>
+        [NotMapped]
+        [Computed]
+        public SiteStatusType Status
+        {
+            get => (SiteStatuses != null ?
```

Comment:
![image](https://user-images.githubusercontent.com/53796823/123195201-9bbf0900-d45c-11eb-86f7-915af22f9aad.png)
im getting this error when trying to run this locally

Reply:
How to reproduce this?

Reply:
FirstOrDefault().StatusType throws if there is nothing in the list.

Reply:
Not sure how I never ran into this lately ... different data?